pkg/protocol/protocol.go

```diff
@@ -189,7 +189,8 @@ func (p *Protocol) shutdown() {
 	p.activeEngineMutex.RLock()
 	p.mainEngine.Shutdown()
 	if p.candidateEngine != nil {
-		p.candidateEngine.Shutdown()
+		// TODO: do we want to delete the candidate engine's data as well?
```

Comment:
remove TODO, this happens automatically on node restart (see enginemanager `LoadActiveEngine`)